engine/collection/pusher/engine.go

```diff
@@ -33,18 +37,44 @@ type Engine struct {
 	state        protocol.State
 	collections  storage.Collections
 	transactions storage.Transactions
+
+	messageHandler *engine.MessageHandler
+	notifier       engine.Notifier
+	inbound        *fifoqueue.FifoQueue
+
+	component.Component
+	cm *component.ComponentManager
 }
 
+// TODO convert to network.MessageProcessor
+var _ network.Engine = (*Engine)(nil)
+var _ component.Component = (*Engine)(nil)
+
 func New(log zerolog.Logger, net network.EngineRegistry, state protocol.State, engMetrics module.EngineMetrics, colMetrics module.CollectionMetrics, me module.Local, collections storage.Collections, transactions storage.Transactions) (*Engine, error) {
+	// TODO length observer metrics
+	inbound, err := fifoqueue.NewFifoQueue(1000)
```

Comment:
I am not sure about the queue capacity we should pick here. I think we should have a good argument (and document it!) of why the number we chose makes sense. If we don't want to go through the trouble of defining a sensible limit, I think we could make this clear by using an _unbounded queue_ ([implementation](https://github.com/onflow/flow-go/blob/master/utils/concurrentqueue/concurrentqueue.go)). 

Anyway, I think a limit makes sense:
* In a nutshell, a collection must reference a block and expires $\texttt{DefaultTransactionExpiry}$ = 600 blocks thereafter https://github.com/onflow/flow-go/blob/a164070da91aacbc92f52599ca0be516a514c4e3/model/flow/constants.go#L18-L22  (its a subtle mechanic, feel free to ignore) 
* consensus nodes produce about $\rho$ = 1.25 blocks per second
* collector clusters produce up to $\varrho$ = 3 collections per second

So very very roughly, between a collection being first proposed (usually referencing the newest known block) to the collection expiring, the collector cluster will produce $\texttt{DefaultTransactionExpiry}\cdot \rho^{-1} \cdot \varrho = 1440$ collections. So if we queued 1500 collections max, then we would be relatively confident that collections have been broadcast before they expire. 
The downside is that collections might sit in the queue for quite some time, if several hundreds are before them. That could be an argument of queuing only much fewer collections (e.g. 200), so collection that don't make it out fast are dropped. 
Another argument we could make is actually using a LIFO queue. If the queue is nearly empty, there is little difference. However, if there is a large backlog of collections to be broadcast, I think its a valid argument of broadcasting the newest first, at least then, some collections make it in time as opposed to all of them being queued past their expiry. Just some thoughts. 

@jordanschalm what do you think?